armory/datasets/package.py

```diff
@@ -0,0 +1,124 @@
+from pathlib import Path
+import os
+import random
+import shutil
+import string
+import subprocess
+
+from armory.logs import log
+from armory.datasets import build, common
+
+
+def package(
+    name,
+    version: str = None,
+    data_dir: str = None,
+    cache_subdir=common.CACHE_SUBDIR,
+    overwrite: bool = False,
+) -> str:
+    """
+    Package a built dataset as .tar.gz, return path
+    """
+    version, data_dir, built_data_dir, subdir = build.build_info(
+        name, version=version, data_dir=data_dir
+    )
+
+    data_dir = Path(data_dir)
+    expected_dir = data_dir / name / version
+    if not expected_dir.is_dir():
+        raise FileNotFoundError(f"Dataset {name} not found at {expected_dir}")
+    tar_full_filepath = common.get_cache_dataset_path(name, version)
+
+    if tar_full_filepath.is_file():
+        if overwrite:
+            tar_full_filepath.unlink(missing_ok=True)
+        else:
+            raise FileExistsError(
+                f"Dataset {name} cache file {tar_full_filepath} exists. Use overwrite=True"
+            )
+
+    log.info("Creating tarball (may take some time)...")
+    cmd = ["tar", "cvzf", str(tar_full_filepath), str(Path(name) / version)]
+    log.info(f"Running {' '.join(cmd)}")
+    completed_process = subprocess.run(cmd, cwd=data_dir)
+    completed_process.check_returncode()
+    return str(tar_full_filepath)
+
+
+def update(name, version: str = None, data_dir: str = None, url=None):
+    """
+    Hash file and update cached datasets file
+    """
+    version, data_dir, built_data_dir, subdir = build.build_info(
+        name, version=version, data_dir=data_dir
+    )
+    filepath = common.get_cache_dataset_path(name, version)
+    if not filepath.is_file():
+        raise FileNotFoundError(f"filepath '{filepath}' not found.")
+    assert (name, version) == common.parse_cache_filename(filepath.name)
+    subdir = str(Path(name) / version)
+    file_size, file_sha256 = common.hash_file(filepath)
+
+    common.update_cached_datasets(name, version, subdir, file_size, file_sha256, url)
+
+
+def verify(name, data_dir: str = None):
+    info = common.cached_datasets()[name]
+    version = info["version"]
+
+    filepath = common.get_cache_dataset_path(name, version)
```

Comment:
should the `data_dir` kwarg be passed along to the `get_cache_dataset_path()` call?

Reply:
Yes, it should.

Also, I'm not the biggest fan of how I pass around `data_dir`, but not sure how we want to structure it.